src/common/timer.c

```diff
@@ -86,8 +86,18 @@ void timesdiv(struct timespec *tvs, COUNTER div)
 }
 
 void
-init_timestamp(timestamp_t *ctx)
+init_timestamp(struct timespec *timestamp)
 {
-    timerclear(ctx);
+    timesclear(timestamp);
 }
 
+int get_current_time(struct timespec *ts){
+    #if defined _POSIX_C_SOURCE  && _POSIX_C_SOURCE >= 199309L
+        return clock_gettime(CLOCK_REALTIME, ts);
```

Comment:
Since the PR also using absolute time semantics for more precise sleeping, both `CLOCK_MONOTONIC`  and `CLOCK_REALTIME` would work. However AFAIK POSIX only defines `CLOCK_REALTIME` mandatory. Probably every sane OS implements both clocksource, but I think realtime chosen because of portability concerns.

Reply:
I'll put it through it's paces when I start work on versin 4.5. I'll see if I can get my confidence up that it won't affect IoT and oldert stuff.